File: awsomepipeline/vpc_stack.py
from aws_cdk import (
    core,
    aws_ec2 as ec2
)

class VpcStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, *, from_vpc_name=None, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        if from_vpc_name is not None:
            self.vpc = ec2.Vpc.from_lookup(self, "vpc", vpc_name=from_vpc_name)
        else:
            self.vpc = ec2.Vpc(self, "vpc", max_azs=2)

File: infrastructure/lib/vpc-stack.ts
import * as cdk from '@aws-cdk/core'
import * as ec2 from '@aws-cdk/aws-ec2'

export class VpcStack extends cdk.Stack {

  public vpc: ec2.IVpc

  constructor(scope: cdk.Construct, id: string, vpcname?: string, props?: cdk.StackProps) {
    super(scope, id, props)

    if (vpcname == undefined) {
      this.vpc = new ec2.Vpc(this, "vpc", { maxAzs: 2 } )
    } else {
      this.vpc = ec2.Vpc.fromLookup(this, "vpc", { vpcName: vpcname })
    }

  }
}
